docs/rules/func-names.md

```diff
@@ -20,6 +20,12 @@ This rule has a string option:
 * `"as-needed"` requires function expressions to have a name, if the name cannot be assigned automatically in an ES6 environment
 * `"never"` disallows named function expressions, except in recursive functions, where a name is needed
 
+This rule has an object option:
+
+* `"generators": true (default) | false`
+    * `true`: require generators to follow the func-names rule. `"always"` or `"as-needed"` will require named generator functions, `"never"` will allow unnamed generator functions.
+    * `false`: require generators to not follow the func-names rule. `"always"` or `"as-needed"` will allow unnamed generator functions, `"never"` will require named generator functions.
+
```

Comment:
The use of `true` and `false` for this case (meaning "use the base option" and "invert the base option") is a bit unintuitive to me. Since there three possible base options (`always`, `as-needed`, and `never`) and only two possible `generators` options, a boolean `generators` option isn't able to express the full space of possible behaviors here. It also seems like the wording of "require generators to follow the `func-names` rule" is self-referential since the option itself is describing what it means for a generator to follow the `func-names` rule.

What would you think about having the `generators` option be a string with the same possible values as the regular option? This seems like it could reduce the cognitive overhead when configuring the option, because the `generators` option would behave the same way as the base option. For example:

```json
{
    "rules": {
        "func-names": ["error", "never", {"generators": "as-needed"}]
    }
}
```

With this scheme, the default behavior of the `generators` option would be to fall back to the base option.

Another possible change could be to make the first option an object, e.g.

```json
{
    "rules": {
        "func-names": [{"base": "never", "generators": "as-needed"}]
    }
}
```

With this setup, it might be clearer how the `generators` and `base` options interact with each other, without reading the documentation. (Note that we would still need to support the string option for backwards compatibility.)

Reply:
Thanks, that makes sense. I'll look at updating the schema as suggested.